File: rb/features.rb
# Woody SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module WoodyFeatures
  def self.make_feature(name)
    case name
    when "base"
      WoodyBaseFeature.new
    when "test"
      WoodyTestFeature.new
    else
      WoodyBaseFeature.new
    end
  end
end

File: py/woody_sdk/features.py
# Woody SDK feature factory

from woody_sdk.feature.base_feature import WoodyBaseFeature
from woody_sdk.feature.test_feature import WoodyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WoodyBaseFeature(),
        "test": lambda: WoodyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
